dashboard/src/components/ColorSchemeToggle.tsx

```diff
@@ -0,0 +1,50 @@
+import IconButton, { IconButtonProps } from '@mui/joy/IconButton';
+import { useColorScheme } from '@mui/joy/styles';
+import * as React from 'react';
+
+import DarkModeRoundedIcon from '@mui/icons-material/DarkModeRounded';
+import LightModeIcon from '@mui/icons-material/LightMode';
+
+export default function ColorSchemeToggle(props: IconButtonProps) {
+    const { onClick, sx, ...other } = props;
+    const { mode, setMode } = useColorScheme();
+    const [mounted, setMounted] = React.useState(false);
```

Comment:
I know it is a minor thing, but would be nice to add the type for the useState.

const [mounted, setMounted] = React.useState<boolean>(false);

Reply:
in that particular case, I decided not to add type since it's becoming boolean automatically. But I'd definitely add a type when it's something a bit more complicated so the person who will be maintaining the project in the future will understand what it's for. What do you think?

Reply:
Sounds good